File: Year 2/Semester 4/MPP/ProtobuffProjectCSharp/model/Artist.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProtobuffProject.domain
{
    [Serializable]
    public class Artist<ID> : Entity<ID>
    {
        private string firstName;
        private string lastName;

        public Artist(ID id, string firstName,
            string lastName) : base(id)
        {
            this.firstName = firstName;
            this.lastName = lastName;
        }

        public string FirstName
        {
            get => this.firstName;
            set => this.firstName = value;
        }

        public string LastName
        {
            get => this.lastName;
            set => this.lastName = value;
        }
        public override string ToString()
        {
            return firstName + " " + lastName;
        }
        
    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 11/REST/RESTJava/Domain/src/main/java/rest/domain/Artist.java
package rest.domain;

import java.io.Serializable;

public class Artist extends Entity<Integer> implements Serializable {
    private String firstName;
    private String lastName;

    public Artist() {
        super(0);
        firstName = "";
        lastName = "";
    }

    public Artist(Integer id, String firstName, String lastName) {
        super(id);
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    @Override
    public String toString() {
        return getId() + " " + firstName + " " + lastName;
    }
}
